app/javascript/components/exercise/CodeBlock.js

```diff
@@ -0,0 +1,85 @@
+import React from 'react'
+import { connect } from 'react-redux'
+
+import AceEditor from 'react-ace'
+import axios from 'axios'
+
+import 'ace-builds/src-noconflict/mode-java'
+import 'ace-builds/src-noconflict/theme-solarized_light'
+
+class CodeBlock extends React.Component {
+  constructor(props) {
+    super(props)
+    this.state = {
+      code: this.props.code
+    }
+    axios.defaults.headers['x-csrf-token'] = document
+      .querySelector('meta[name="csrf-token"]')
+      .getAttribute('content')
+  }
+
+  onChange = newValue => {
+    this.setState({ code: newValue })
+  }
+
+  handleSubmit = async () => {
+    //console.log('code', this.state.code)
+    // request to backend for testing
+    const result = await axios
+      .post(
+        '/api/submissions/question',
+        {
+          /*
+          file_name: this.props.file,
+          code: this.state.code
+          */
+          "id": 6,
+	        "file_name": "/src/main/java/sequence/NumSequenceGenerator.java",
+	        "code":
```

Comment:
**[eslint]** <sub>reported by [reviewdog](https://github.com/haya14busa/reviewdog) :dog:</sub>
Replace `↹········"code":⏎↹········{⏎↹↹········1:·"········NumData[]·seq·=·new·NumData[size];\n········for·(int·i·=·0;·i·<·size;·i++)·{\n············seq[i]·=·new·NumData(i);\n········}\n········return·new·NumDataSequence(seq);"` with `··········code:·{⏎············1:·'········NumData[]·seq·=·new·NumData[size];\n········for·(int·i·=·0;·i·<·size;·i++)·{\n············seq[i]·=·new·NumData(i);\n········}\n········return·new·NumDataSequence(seq);'` (prettier/prettier)